public/v4/apps/docker-registry.yml

```diff
@@ -0,0 +1,40 @@
+captainVersion: 4
+services:
+    $$cap_appname-registry:
+        image: registry:$$cap_registry_version
+        volumes:
+            - $$cap_appname-registry-data:/var/lib/registry
+            - $$cap_appname-registry-auth:/auth
+        environment:
+            REGISTRY_AUTH: htpasswd
+            REGISTRY_AUTH_HTPASSWD_PATH: /auth/htpasswd
+            REGISTRY_AUTH_HTPASSWD_REALM: Docker Registry
+        caproverExtra:
+            containerHttpPort: '5000'
+    $$cap_appname-ui:
+        depends_on:
+            - $$cap_appname-registry
+        image: joxit/docker-registry-ui:static
+        restart: always
+        environment:
+            DELETE_IMAGES: 'true'
+            REGISTRY_URL: http://srv-captain--$$cap_appname-registry:5000
+        caproverExtra:
+            containerHttpPort: '80'
+caproverOneClickApp:
+    variables:
+        - id: $$cap_registry_version
+          label: Docker Registry image version
+          defaultValue: 2
+          description: Check out their Docker page for the valid tags https://hub.docker.com/_/registry?tab=tags - default is 2 as of 2020-09-20
+    instructions:
+        start: Standard docker registry with a Web UI for image maintenance.
+
+            Make sure you create a htpasswd file somewhere on your host with htpasswd -Bbn username password > /<path>/htpasswd.
```

Comment:
This is a bit complex. Can we, instead of a plain Dockerfile, we use dockerfileLines, and just create the password during container start up?

Reply:
Let me investigate how this is done properly and I'll submit a new PR. Thanks for the suggestion.

Reply:
Could be something like:
```
FROM registry:$$cap_registry_version
RUN htpasswd -Bbn $$cap_username $$cap_password > /auth/htpasswd
```
Also, you'd have to remove `$$cap_appname-registry-auth:/auth` volume as it'll override it.